Tools/TestWebKitAPI/Tests/WebKitGLib/TestWebKitWebView.cpp

```diff
@@ -1411,6 +1412,39 @@ static void testWebViewIsWebProcessResponsive(WebViewTest* test, gconstpointer)
     g_assert_true(webkit_web_view_get_is_web_process_responsive(test->m_webView));
 }
 
+static void testWebViewIsWebProcessUnresponsive(WebViewTest* test, gconstpointer)
+{
+    static const char* hangHTML =
+        "<html>"
+        " <body>"
+        "  <script>"
+        "   setTimeout(function() {"
+        "    var start = new Date().getTime();"
+        "    var end = start;"
+        "     while(end < start + 4000) {"
+        "      end = new Date().getTime();"
+        "     }"
+        "    }, 10);"
+        "  </script>"
+        " </body>"
+        "</html>";
+
+    g_assert_true(webkit_web_view_get_is_web_process_responsive(test->m_webView));
+    test->loadHtml(hangHTML, nullptr);
+    test->waitUntilLoadFinished();
+    // Wait 20 milliseconds, so the js while loop kicks in and blocks the web process. Then try to load a new
+    // page. As the web process is busy this won't work, and after 3 seconds the web process will be marked
+    // as unresponsive.
+    test->wait(0.02);
```

Comment:
The 0.02 is just enough period of time to the initial start of the responsivenessTimer happens. Then we do a restart the responsivenessTimer because the initial one does not catch the "js while loop". 
3sec is the default value after the process becomes considered as unresponsive.
https://github.com/WebKit/WebKit/blob/main/Source/WebKit/UIProcess/AuxiliaryProcessProxy.h#L51
https://github.com/WebKit/WebKit/blob/main/Source/WebKit/UIProcess/ResponsivenessTimer.h#L50